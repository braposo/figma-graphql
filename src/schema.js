const { makeExecutableSchema } = require("graphql-tools");
const { GraphQLDateTime } = require("graphql-iso-date");
const mergeSchema = require("./types");

const typeDefs = [
    `
    scalar DateTime

    type Query {
        version: String!
    }

    type Mutation {
        noop: String
    }
    
    type Subscription {
        noop: String
    }
    
    enum BlendMode {
        PASS_THROUGH
        NORMAL
        DARKEN
        MULTIPLY
        LINEAR_BURN
        COLOR_BURN
        LIGHTEN
        SCREEN
        LINEAR_DODGE
        COLOR_DODGE
        OVERLAY
        SOFT_LIGHT
        HARD_LIGHT
        DIFFERENCE
        EXCLUSION
        HUE
        SATURATION
        COLOR
        LUMINOSITY
    }
    
    enum StrokeAlign {
        INSIDE
        OUTSIDE
        CENTER
    }
`,
];

const resolvers = {
    DateTime: GraphQLDateTime,
    Query: {
        version: () => "1.4",
    },
};

module.exports = {
    schema: makeExecutableSchema(mergeSchema({ typeDefs, resolvers })),
    context: () => ({}),
};
