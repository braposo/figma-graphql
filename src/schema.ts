import { makeExecutableSchema } from "graphql-tools";
import { GraphQLDateTime } from "graphql-iso-date";
import mergeSchema from "./types";

const typeDefs = [
    `
    scalar DateTime

    type Query {
        version: String!
    }

    type Mutation {
        noop: String
    }
    
    type Subscription {
        noop: String
    }
`,
];

const resolvers = {
    DateTime: GraphQLDateTime,
    Query: {
        version: () => "1.4",
    },
};

export default makeExecutableSchema(mergeSchema({ typeDefs, resolvers }));
