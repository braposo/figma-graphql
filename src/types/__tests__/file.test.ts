/* eslint-env jest */

import { graphql } from "graphql";
import schema from "../../schema";

const figmaFile = "cLp23bR627jcuNSoBGkhL04E";

jest.setTimeout(10000);

describe("File", () => {
    test("can get file properties", async () => {
        const query = `
            query {
                file(id: "${figmaFile}") {
                    name
                    lastModified
                    thumbnailUrl
                    version
                }
            }
        `;

        const response = await graphql(schema, query);
        const { name, lastModified, thumbnailUrl, version } = response.data && response.data.file;

        expect(name).toEqual("figma-graphql test file");
        expect(new Date(lastModified)).toEqual(expect.any(Date));
        expect(thumbnailUrl).not.toBeNull();
        expect(version).not.toBeNull();
    });

    test("can get default file images", async () => {
        const query = `
            query {
                file(id: "${figmaFile}") {
                    images {
                        id
                    }
                }
            }
        `;

        const response = await graphql(schema, query, null, { fileId: figmaFile });

        expect(response).toEqual({
            data: { file: { images: [{ id: "0:1" }] } },
        });
    });

    test("can get specific file images", async () => {
        const query = `
            query {
                file(id: "${figmaFile}") {
                    images(params: { ids: ["1:6"]}) {
                        id
                    }
                }
            }
        `;

        const response = await graphql(schema, query, null, { fileId: figmaFile });

        expect(response).toEqual({
            data: { file: { images: [{ id: "1:6" }] } },
        });
    });

    test("can get file comments", async () => {
        const query = `
            query {
                file(id: "${figmaFile}") {
                    comments {
                        id
                        message
                        file_key
                        order_id
                        parent_id
                        created_at
                        client_meta {
                            node_id
                            node_offset {
                                x
                                y
                            }
                        }
                        resolved_at
                      }
                }
            }
        `;

        const response = await graphql(schema, query, null, { fileId: figmaFile });

        expect(response).toMatchSnapshot();
    });
});
