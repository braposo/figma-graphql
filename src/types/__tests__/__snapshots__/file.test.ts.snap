// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File can get default file exports 1`] = `
Object {
  "data": Object {
    "file": Object {
      "exports": Array [
        Object {
          "id": "0:1",
        },
      ],
    },
  },
}
`;

exports[`File can get file comments 1`] = `
Object {
  "data": Object {
    "file": Object {
      "comments": Array [
        Object {
          "client_meta": Object {
            "node_id": "28:5",
            "node_offset": Object {
              "x": 111,
              "y": 22,
            },
          },
          "created_at": "2019-05-26T17:47:36Z",
          "file_key": "cLp23bR627jcuNSoBGkhL04E",
          "id": "6907475",
          "message": "Testing comment",
          "order_id": 1,
          "parent_id": "",
          "resolved_at": null,
        },
      ],
    },
  },
}
`;

exports[`File can get specific file exports 1`] = `
Object {
  "data": Object {
    "file": Object {
      "exports": Array [
        Object {
          "id": "1:6",
        },
      ],
    },
  },
}
`;
